testing/async/inclusive_scan/using_vs_adl.cu

```diff
@@ -0,0 +1,169 @@
+#include <thrust/detail/config.h>
+
+#if THRUST_CPP_DIALECT >= 2014
+
+#include <async/test_policy_overloads.h>
+
+#include <async/inclusive_scan/mixin.h>
+
+// Verify what happens when calling the algorithm without any namespace
+// qualifiers:
+// - If the async entry point is available in the global namespace due to a
+//   using statement, the async algorithm should be called.
+// - Otherwise, ADL should resolve the call to the synchronous algo in the
+//   thrust:: namespace.
+
+namespace invoke_reference
+{
+
+template <typename input_value_type,
+          typename output_value_type = input_value_type>
+struct adl_host_synchronous
+{
+  template <typename InputType,
+            typename OutputType,
+            typename PostfixArgTuple,
+            std::size_t... PostfixArgIndices>
+  static void invoke_reference(InputType const& input,
+                               OutputType& output,
+                               PostfixArgTuple&& postfix_tuple,
+                               std::index_sequence<PostfixArgIndices...>)
+  {
+    // Create host versions of the input/output:
+    thrust::host_vector<input_value_type> host_input(input.cbegin(),
+                                                     input.cend());
+    thrust::host_vector<output_value_type> host_output(host_input.size());
+
+    using OutIter = thrust::remove_cvref_t<decltype(host_output.begin())>;
+
+    // ADL should resolve this to the synchronous `thrust::` algorithm.
+    // This is checked by ensuring that the call returns an output iterator.
+    OutIter result =
+      inclusive_scan(host_input.cbegin(),
+                     host_input.cend(),
+                     host_output.begin(),
+                     std::get<PostfixArgIndices>(THRUST_FWD(postfix_tuple))...);
```

Comment:
I don't think synchronous scans are marked nodiscard, so I think we can just ignore the result altogther.

Reply:
Good point, I got carried away following the reduce algorithms as a reference :) I've removed the nodiscard annotations completely.